sympy/functions/special/tests/test_bessel.py

```diff
@@ -69,6 +69,14 @@ def test_besselk_leading_term():
     assert besselk(0, x).as_leading_term(x) == -log(x) - S.EulerGamma + log(2)
```

Comment:
We need to test all cases extensively specially for the leading term method, otherwise it'll be pretty easy to mess up simple limits.
For example this is one I could find out from just trying 
```
>>> besselk(0, cos(x)).as_leading_term(x)
besselk(0, cos(x))    # should be besselk(0, 1)
>>> besselk(0, cos(x)).series(x, n = 4)
besselk(0, 1) + x**2*besselk(1, 1)/2 + O(x**4)
```
The series looks perfectly fine though

Reply:
cc @arnabnandikgp I would really you to help @Kishore96in with the tests here. No harm in doing in extensive testing because as far as I know we've had poor output for limits/series for bessel functions (especially the modified ones)

It would help with your PR too.

Reply:
This is the `e.is_zero` cases (after going through the positive and negative) case

Reply:
@anutosh491 I have now fixed the `e.is_zero` case. Also added tests for this and the e<0 case.

```
In [1]: besselk(1,cos(x)).as_leading_term(x)
Out[1]: besselk(1, 1)

In [2]: besselk(3,1/x).as_leading_term(x)
Out[2]: 
       -1 
       ───
        x 
√2⋅√π⋅ℯ   
──────────
      ___ 
     ╱ 1  
2⋅  ╱  ─  
  ╲╱   x  

In [9]: besselk(3,1/sin(x)).as_leading_term(x)
Out[9]: 
       -1 
       ───
        x 
√2⋅√π⋅ℯ   
──────────
      ___ 
     ╱ 1  
2⋅  ╱  ─  
  ╲╱   x  


```
Unsure about the last one, since it disagrees with @arnabnandikgp 's implementation of `besselk._eval_aseries`
```In [8]: besselk(3,1/sin(x)).series(x, n=1).expand().removeO()
Out[8]: 
        -1    
       ────── 
       sin(x) 
√2⋅√π⋅ℯ       
──────────────
      ________
     ╱   1    
2⋅  ╱  ────── 
  ╲╱   sin(x)
```

Reply:
Yess this is an error. Let's raise an issue for the same

Reply:
I am satisfied with the tests added in the recent commit.